content/gotraining-2019-05-melbourne.slide

```diff
@@ -645,6 +645,28 @@ If `master` on upstream has advanced in the meantime:
 	}
 
 
+* Embedding - Structs
+
+"When we embed a type, the methods of that type become methods of the outer type, but when they are invoked the receiver of the method is the inner type, not the outer one." - [[https://golang.org/doc/effective_go.html#embedding][Effective Go]]
+
+- The same applies for all fields within an embedded type
+
+
+* Precedence when Embedding
+
+Embedding types introduces the possiblity for naming conflicts. Go follows some simple rules to help resolve this:
+
+- A field or method "X" will hide any other field or method "X" found in a more deeply nested part of the type
+- If a field or method appears at the same nesting level, it will return an error
+
+
+* Examples
+
+- [[https://goplay.space/#t4oJw-_2mph][Embedded Structs]]
+- [[https://goplay.space/#T3wiMPCEd_x][Embedded Interfaces]]
+- [[https://goplay.space/#Vw-xfHsNufu][Precedence]]
```

Comment:
Couple of things: no need to embed via pointers in these examples I think.
LastName isn't used I think.
Sample doesn't compile, better comment out the bits that break the code IMO.

Suggestion:
https://goplay.space/#3BUfQW3MxG9

Reply:
Cool example, I didn't realise you could specify the exact field you wanted to extract in the embedded struct like that if there was a conflicting name at a deeper nested level.